### ISSUE TITLE
[papyrus 2.0] Setup git subtree

### ISSUE THREAD

**Issue by svx** [plone/papyrus#143]
We should setup a git subtree to include the new image repository.

With that we should get the latest image updates from jenkins during git pull.

Steps to do:

- [ ] Add the steps to install/setup part in the docs
- [ ] Do the setup

@polyester is that about right ? Or do I forgot something ?

**Comment by polyester** [comment]
Probably decide on a sane branching/naming scheme for https://github.com/plone/documentation-roboshots

Currently "master" there is latest 5.1, but hand-checked to know they are good _(well, as good as robot-generated can be)_

If we start to do Jenkins-generated ones, they should probably go into another branch, to avoid the situation that a borked test, or a breakage in the headless browser used, wipes them all out. So maybe use this scheme:

- master: last known good pics from current coredev.buildout master
- 5.1: 'frozen' 5.1 release when 5.2 becomes master
- testing: always the latest shots made by jenkins; if they pass manual review, they can be moved to master. _"manual review" meaning: are they all blank, or are large parts of them pure black. I've seen both with various browser updates_

Of course that's open for discussion, but I want to avoid that we pull in broken screenshots when the latest jenkins-run has issues, and robot-screenshots are still brittle based on browser upgrades.

Then, we would pull in 'master' as git subtree in papyrus 2.0, I'd say

**Comment by svx** [comment]
What is with the idea to do it like RTD ? Would that OK?

Meaning:

latest = always the latest aka unstable/testing [would be master]
and below that we have branches according to versions.

That is basically the same what I would like to do with the docs.

Does that would make sense ?

**Comment by polyester** [comment]
Makes sense, only I would like to have a 'last known good' set of pics. Although I suppose you could do a git revert if one run breaks everything, but that is a hassle.

Still, maybe it's good having a setup where the Jenkins-based testruns first go to a 'staging-5.1' or 'staging-5.2' branch before being put in "latest" and the 5.x branches.  At least, until the ecosystem of "headless browsers + webdrivers" is rock-solid. At the moment it is simply not.

Since we don't have Jenkins-based testruns yet, this is a bit academic, but it would make me feel comfortable to know we always have a 'known good set' of pics to offer end users and ourselves.
